x/distribution/keeper/delegation.go

```diff
@@ -136,6 +136,16 @@ func (k Keeper) CalculateDelegationRewards(ctx sdk.Context, val stakingtypes.Val
 	return rewards
 }
 
+// SmartRewardAccount is an account with a post-reward processing function.
+// Such an account ignores a non-default withdrawal address, as this can
+// be implemented in the post-reward processing if desired.
+type SmartRewardAccount interface {
```

Comment:
why not introduce Distribution Hooks instead of a new account?

Reply:
See below.